### PR TITLE
[wip] add more ntac tactics

### PR DESCRIPTION
**Pull request by stchang**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cur [![Build Status](https://travis-ci.org/wilbowma/cur.svg?branch=master)](https://travis-ci.org/wilbowma/cur)
+cur [![Build Status](https://travis-ci.org/stchang/cur.svg?branch=ntac-more-tactics)](https://travis-ci.org/stchang/cur)
```
Comment (stchang) [inline README.md]: oops, will revert this

File: cur-lib/cur/ntac/coq-rewrite-old.rkt [hunk 2]
```diff
@@ -0,0 +1,176 @@
+#lang s-exp "../main.rkt"
```
Comment (stchang) [inline cur-lib/cur/ntac/coq-rewrite-old.rkt]: will delete this too

File: cur-lib/cur/ntac/standard.rkt [hunk 3]
```diff
@@ -150,18 +184,321 @@
...
+       #`(fill (destruct #'x #'param-namess))]))
+
+  ;; (struct identifier-info (type delta-def))
+  ;; ;; TODO PERF: Could use vectors instead of lists; since we store the count anyway... or maybe we won't
+  ;; ;; need to by storing param and index decls separately.
+  ;; (struct constant-info identifier-info
+  ;;   (param-count param-name-ls param-ann-ls index-name-ls index-ann-ls
+  ;;                constructor-count constructor-ls constructor-index
+  ;;                recursive-index-ls))
+
```
Comment (stchang) [inline cur-lib/cur/ntac/standard.rkt]: agh, forgot to delete this debugging code

**Comment by wilbowma** [comment]
I'm refactoring and integrating these gradually in branch pr-79. I've refactored a bunch, and now Basics.rkt isn't passing. Seems like a problem with one of the rewrite tactics, which transforms the goal `(== Nat (plus n n) (plus m m))` to the goal `m`.

Could be that I introduced a bug when I integrated some of the stxutils into master... not sure what else it could be, if they were passing before.

**Comment by wilbowma** [comment]
Okay, pr-79 is refactored more or less the way I like it. If I were being super obsessive, I'd try to split the standard tactics and rewrite tactics into separate commits... but their tests are a bit too interdependent.

Before I merge this, I'd like documentation for the new tactics, though.

**Pull request closed without merging**